crypto/rand/drbg_lib.c

```diff
@@ -380,16 +427,51 @@ int RAND_DRBG_instantiate(RAND_DRBG *drbg,
         max_entropylen += drbg->max_noncelen;
     }
 
+    /*
+     * If we're running with continuous RNG testing and haven't initialised
+     * yet, we will need more data.  We add one block and round up to the
+     * nearest block boundary.  The entropy requirements are not altered.
+     */
+    if (bs > 0) {
```

Comment:
To me it feels wrong and hacky to integrate the CRNG code into the DRBG implementation itself. In particular this tweaking of the input parameters gives me a bad feeling. In my opinion the design would be cleaner if the CRNG test would be implemented as a filter between the DRBG and its entropy source: It takes the request from the DRBG, requests the extra amount of random data from the source and tests the return data before handing it over to the DRBG. In other words, the testing should happen transparently to the DRBG.

Currently I am also pondering whether this extra block should always be discarded and not fed into the DRBG.

Reply:
NIST mandates the discard of the first block, so we'll need to be able to do it.
It does seem like a loss though.